### DISCUSSION TITLE
"matchPackageNames" vs "matchDepName" - both seem to use "depName" anyway

### DISCUSSION THREAD

**Discussion by szpak** [renovatebot/renovate#20416] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.129.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Question

After empirical tests `matchPackageNames` uses `depName` property. `matchDepName` probably as well. What is a difference between those two `matchers`?

I ask, as I've just spent some time trying to group dependencies from the `maven-wrapper` manager  where in addition `packageName` property is used in the dependency definition and I was surprised that `matchPackageNames` uses `depName`, not `packageName` to match :). If that the intention, maybe it would be good to mention that in the `matchPackageNames` description (that `depName` is used)?

Btw, in the end I used "matchManagers", but the question is still valid.

### Relevant debug logs

<details><summary>Logs</summary>

Sample log file for maven-wrapper dependencies (`depName` and `packageName`):

```
    "maven-wrapper": [
      {
        "packageFile": ".mvn/wrapper/maven-wrapper.properties",
        "deps": [
          {
            "depName": "maven",
            "packageName": "org.apache.maven:apache-maven",
            "currentValue": "3.8.6",
            "replaceString": "https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.6/apache-maven-3.8.6-bin.zip",
            "datasource": "maven",
            "versioning": "maven",
            "depIndex": 0,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "3.9.0",
                "newValue": "3.9.0",
                "releaseTimestamp": "2023-01-31T10:24:38.000Z",
                "newMajor": 3,
                "newMinor": 9,
                "updateType": "minor",
                "branchName": "renovate/maven-3.x"
              }
            ],
            "warnings": [],
            "registryUrl": "https://repo.maven.apache.org/maven2",
            "homepage": "https://maven.apache.org/",
            "currentVersion": "3.8.6",
            "isSingleVersion": true,
            "fixedVersion": "3.8.6"
          },
...


</details>

**Comment by viceice** [comment, marked as answer]
matchPackageNames historically matches depName, but should match packageName instead. this will be changed in a future major version.

you can already switch the behavior on self-hosted renovate via an experimental option 


https://docs.renovatebot.com/self-hosted-experimental/#renovate_x_match_package_names_more

**Reply by szpak** [reply]
> this will be changed in a future major version.

I see. What will be then matched by  `matchPackageNames` - for example - for Maven dependencies (regular, not from maven-wrapper) which don't have `packageName` at all (just `depName`)?

**Reply by rarkins** [reply]
`packageName` is always the real package name. `depName` is used sometimes as the "short" name. e.g. `depName` might be `node` but `packageName` is `nodejs/node` (the repo)

**Reply by szpak** [reply]
Thanks. One more question. As in some managers the `packageName` is currently not available (e.g. "maven", not "maven-wrapper"), the `matchPackageNames` (in the new version) will fallback to `depName` or the managers model will be enhanced to have `packageName` as well?

```json
          {
            "datasource": "maven",
            "depName": "io.swagger.core.v3:swagger-annotations",
            "currentValue": "2.2.8",
            "fileReplacePosition": 6576,
            "registryUrls": [
              "https://repo.maven.apache.org/maven2/",
            ],
            "depType": "compile",
            "depIndex": 9,
            "updates": [],
            "warnings": [],
            "versioning": "maven",
            "sourceUrl": "https://github.com/swagger-api/swagger-core",
            "currentVersion": "2.2.8",
            "fixedVersion": "2.2.8"
          },
```

**Reply by rarkins** [reply]
Yes, it will fall back
